monai/handlers/mlflow_handler.py

```diff
@@ -115,14 +183,61 @@ def attach(self, engine: Engine) -> None:
             engine.add_event_handler(Events.ITERATION_COMPLETED, self.iteration_completed)
         if self.epoch_log and not engine.has_event_handler(self.epoch_completed, Events.EPOCH_COMPLETED):
             engine.add_event_handler(Events.EPOCH_COMPLETED, self.epoch_completed)
+        if not engine.has_event_handler(self.complete, Events.COMPLETED):
+            engine.add_event_handler(Events.COMPLETED, self.complete)
 
-    def start(self) -> None:
+    def start(self, engine: Engine) -> None:
         """
         Check MLFlow status and start if not active.
 
         """
+        mlflow.set_experiment(self.experiment_name)
         if mlflow.active_run() is None:
-            mlflow.start_run()
+            run_name = f"run_{time.strftime('%Y%m%d_%H%M%S')}" if self.run_name is None else self.run_name
+            mlflow.start_run(run_name=run_name)
+
+        if self.experiment_param:
+            # avoid to recording same parameters in the same run
+            self._delete_exist_param_in_dict(self.experiment_param)
+            if self.experiment_param:
+                mlflow.log_params(self.experiment_param)
+
+        attrs = {attr: getattr(engine.state, attr, None) for attr in self.default_attr_name}
+        self._delete_exist_param_in_dict(attrs)
+        mlflow.log_params(attrs)
+
+        for param_name in self.default_log_param_list:
+            if self._is_param_exists(param_name):
+                continue
+            self._try_log_param(engine, param_name)
+
+    def _parse_artifacts(self):
+        """
+        Log artifacts to mlflow. Given a path, all files in the path will be logged recursively.
+        Given a file, it will be logged to mlflow.
+        """
+        artifact_list = []
+        for path_name in self.artifacts:
+            if os.path.isfile(path_name):
+                artifact_list.append(path_name)
+            else:
+                for root, _, filenames in os.walk(path_name):
+                    for filename in filenames:
+                        file_path = os.path.join(root, filename)
+                        artifact_list.append(file_path)
+        return artifact_list
+
+    def complete(self) -> None:
+        """
+        Handler for train or validation/evaluation completed Event.
+        """
+        # If just input one path, convert it to list.
+        if isinstance(self.artifacts, str):
+            self.artifacts = [self.artifacts]
```

Comment:
Please avoid to changing member variable during training, I would suggest to set it in the `__init__()`:
```py
self.artifacts = ensure_tuple(artifacts)
```